레이아웃 및 GNB 구현

seongjin2427 opened the pull request:
# Pull Request

- [x] Feature

## Summary

- 메인 레이아웃 구현
- 메인 레이아웃의 GNB(Header) 및 네비게이션  구현

## Describe your changes

- Layout
  - fullwidth Props 유무로 max-width 1280px 적용을 지정할 수 있음
  - `<MainLayout>` max-width 1280px 미 적용
  - `<MainLayout fullWidth>` max-width 1280px 적용

- GNB (Header)
  - 태블릿 크기 `(SCREEN_TABLET)`에서부터 HamburgerButton 활성화 및 네비게이션 창 적용
  - Logo 및 검색 아이콘 SVG 파일 및 컴포넌트 추가

- 미디어 쿼리 디바이스별 상수 정의
  - `SCREEN_TABLET = '@media (min-width: 768px)';`
  - `SCREEN_DESKTOP = '@media (min-width: 1280px)';`
- HamburgerButton styled-component화 완료
- 로그인 유무에 따른 로그인/회원가입 ↔ 로그아웃/마이페이지 버튼 전환

## Issue number and link

- 이슈 #25 

## ETC

리덕스 적용 후 GNB의 isLogin Props 연결 필요

seongjin2427 committed:
Merge branch 'develop' of https://github.com/TRIPONG-2022/frontend into develop

seongjin2427 committed:
Merge branch 'develop' of https://github.com/TRIPONG-2022/frontend into develop

seongjin2427 committed:
feat: GNB 1차 구현 및 MainLayout 1차 구현

- GNB 컴포넌트 추가
- MainLayout 추가
- 뷰포트 확인을 위한 useWindowSize hook 추가
- useEffect시 마지막 이벤트에 대해서만 확인하도록 deboundFunction 추가
- MainLayout Fullwidth Props로 빼면 Max-width 1280px 적용 가능
- 반응형 네비바 구현

seongjin2427 committed:
Merge branch 'develop' of https://github.com/TRIPONG-2022/frontend into feature/layout

seongjin2427 committed:
feat: XButton 추가

- XButton 1차 구현

seongjin2427 committed:
feat: GNB 수정

- hook -> hooks 폴더명 수정
- XButton -> Hamburger로 컴포넌트명 수정
- HamburgerButton 클릭 시 나오는 창 별도 분리

seongjin2427 committed:
fix: HamburgerButton Styled-component화

- HamburgerButton의 기존 css 클래스명 작동방식에서 styled-component를
  활용하여 변경

seongjin2427 committed:
fix: 반응형 태블릿 버전 추가 및 GNB 일부 수정

- 모바일 버전 (768px padding: 20px)
- 태블릿 버전 추가 (768px~1024px padding: 32px)
- 데스크탑 버전 (1025px~ padding: 40px)
- 윈도우창 리사이즈 시 네비바 딸려오는 버그가 있어서 수정
  - transformX(100vw) -> transformX(150vw) 변경

seongjin2427 committed:
fix: 임시저장

seongjin2427 committed:
feat: Logo, Search 아이콘 추가, GNB 내 NavigationDiv 분리

- Logo, Search SVG 파일 및 아이콘 컴포넌트 추가
- GNB 내부 NavigationDiv 컴포넌트 분리

seongjin2427 committed:
Merge branch 'develop' of https://github.com/TRIPONG-2022/frontend into feature/layout

seongjin2427 committed:
refact: 미디어 쿼리 상수 추가 (screen.ts)

- 미디어 쿼리 상수 추가
  - Mobile (max-width: 768px)
  - Tablet (max-width: 1024px)
  - Desktop (min-width: 1025px)
- 상수 추가에 따른 styled-component 수정

sangrimlee commented:
branch가 main branch로 되어있는데 수정해주셔야 될 것 같아요.

seongjin2427 commented:
으얽 변경했습니다 ㅠㅠ

jma1020 reviewed (approved):
확인했습니다!! 완전 Good 코드입니다 😊

seongjin2427 committed:
refact: Layout 및 GNB 모바일 퍼스트 미디어 쿼리로 변경

- 모바일 퍼스트로 변경
- 변경에 따른 styled-component 스타일 수정
- 네비바 토글을 위한 토글 초기값 false로 변경

sangrimlee commented on styles/screen.ts:
```diff
@@ -0,0 +1,2 @@
+export const SCREEN_MOBILE = '@media (min-width: 768px)';
```
`SCREEN_MOBILE`을 `SCREEN_TABLET`으로 `SCREEN_TABLET`을 `SCREEN_DESKTOP`으로 변경해야 될 것 같아요.

seongjin2427 replied:
반영 완료했습니닷! 😁

sangrimlee commented on layouts/NavigationDiv/NavigationDiv.tsx:
```diff
@@ -0,0 +1,35 @@
+  isLogin: boolean;
+}
+
+function NavigationDiv({ toggle, isLogin }: NavigationDivProps) {
```
`NavigationDiv`보다는 `MobileNavigation`이라던지 조금 더 명확한 이름이 좋지 않을까요??

seongjin2427 replied:
저도 좀 더 명확한 이름으로 하고 싶었는데, MobileNavigation으로 변경했습니다! 감사합니다!

sangrimlee commented on layouts/MainLayout/MainLayout.styled.ts:
```diff
@@ -0,0 +1,44 @@
+    `}
+`;
+
+export const Footer = styled.div`
```
`div`보다는 `footer`를 쓰는게 Semantic 하지 않을까요?
이 부분 말고도  Body 부분도 `main`을 GNB에서의 ContainerNav는 `nav`태그를 `header`태그로 한 번 감싸서 쓰는게 더 semantic 할 것 같아요!

seongjin2427 replied:
Sementic하게 작성해야 한다는것을 알고는 있었는데, 프로젝트에서는 크게 신경을 안 쓰고 있었네요!
피드백 감사합니다!
바로 반영했어요!

sangrimlee commented on components/shared/HamburgerButton/HamburgerButton.tsx:
```diff
@@ -0,0 +1,29 @@
+// import styles from './HamburgerButton.module.css';
+import * as Styled from './HamburgerButton.styled';
+
+interface XButtonProps {
```
HamburgerButtonProps로 바꿔줘야 될 것 같아요:)

seongjin2427 replied:
핫 수정 완료 했슴당

seongjin2427 committed:
refact: HamburgerButton Props명 변경

- XButtonProps -> HamburgerProps

seongjin2427 committed:
Merge branch 'develop' of https://github.com/TRIPONG-2022/frontend into feature/layout

seongjin2427 committed:
refact: feedback 반영

- 미디어 쿼리 contants명 변경
  - SCREEN_MOBILE -> SCREEN_TABLET
  - SCREEN_TABLET -> SCREEN_DESKTOP
- NavigationDiv 컴포넌트명 명확하게 변경 (-> MobileNavigation)
- 시멘틱 태그 사용
  - MainLayout
    - Container -> <body>
    - GNB -> <header>
    - Body -> <main>
    - Footer -> <footer>

seongjin2427 committed:
refact: MainLayout - Container 태그  body -> div 변경

- styled-component에서 body로 지정하니까 오류가 나서 div로 변경

sangrimlee reviewed (approved):
고생하셨어요👍
